compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2783,6 +2794,38 @@ public void visit(BLangErrorConstructorExpr errorConstructorExpr) {
         resultType = errorConstructorExpr.type = errorConstructorType;
     }
 
+    private BSymbol findMemberWithMatchingErrorType(BUnionType expType, BLangErrorConstructorExpr errorCtor) {
+        List<BLangNamedArgsExpression> providedArgs = getProvidedErrorDetails(errorCtor);
+        List<BType> providedFieldsType = providedArgs.stream().map(f -> f.expr.type).collect(Collectors.toList());
```

Comment:
Shouldn't we decide on the expression types based on the contextually expected types? There seems to be an issue with how we type check the detail args anyway - https://github.com/ballerina-platform/ballerina-lang/issues/29133.

Reply:
Agree that field type should be decided based on contextually expected type, not sure if this is a good time to spend the time.
@hasithaa should we fix #29133 here, or fix after alpha?

Reply:
Decided to mark #29133 a blocker in a meeting as it's a spec deviation.